### PR TITLE
firmware: Optimize code size by using goto, saving 59 bytes

### PR DESCRIPTION
**Pull request by purdeaandrei**
Saves 59 bytes

### CODE REVIEW

**Comment by whitequark** [comment]
I think you could probably get the same benefit by converting that giant conditional to be an `if..else` one and adding explicit `return` to all paths that do not stall, leaving the stalling paths to fall through.

**Comment by purdeaandrei** [comment]
When you mean giant conditional, are you referring to this?:
```c
if(glasgow_config.revision < GLASGOW_REV_C0 ||
         !iobuf_get_pull(arg_selector,
                         (__xdata uint8_t *)EP0BUF + 0,
                         (__xdata uint8_t *)EP0BUF + 1)) {
```

**Comment by whitequark** [comment]
Yeah.

**Comment by purdeaandrei** [comment]
hmm, the two stalls in the while loop: I don't think those can be optimized without goto, unless I add another check to see if the loop exited early or not

**Comment by purdeaandrei** [comment]
I tried without goto (except for those two stalls in the while loop), and the code size increased!

**Comment by purdeaandrei** [comment]
This is what I tried: 
https://github.com/purdeaandrei/glasgow/commit/85deafc2264b8a7cc7ad7a716e248e1f1467d84f

And it increased code size by 37 bytes :( . I don't know what the explanation is

**Comment by whitequark** [comment]
Oh no ;; sdcc is like this sometimes. Thank you for trying!

In this case the `goto` solution is the obvious winner, let me review this.

File: firmware/main.c [hunk 1]
```diff
@@ -783,6 +779,8 @@ void handle_pending_usb_setup() {
     }
 
     return;
+stall_ep0_return:
+    STALL_EP0();
```
Comment (whitequark) [inline firmware/main.c:783]:
```suggestion
    return;

    // Factor out the stall exit to reduce code size.
stall_ep0_return:
    STALL_EP0();
```

**Review by whitequark — APPROVED** [review]
LGTM, thanks!

**Pull request merged**